### DISCUSSION TITLE
How to translate words in the component(.ts) file??

### DISCUSSION THREAD

**Discussion by mouad-mahfoud** [ng-alain/ng-alain#2298] [Q&A]
At present I am using the translation in ".html" file. But, now I have to use the translation in "Component(.ts)" file as well.

**Comment by cipchk** [comment]
Pls refer to [$localize](https://angular.io/api/localize/init/$localize)

**Reply by mouad-mahfoud** [reply]
I use this in my components class 
`this.nzI18nService.translate('app.anchor.definition')`

but it not work

**Reply by cipchk** [reply]
`NzI18nService` just serves the `ng-zorro-antd` class library (`@delon/*` also has a similar i18n class `DelonLocaleService`, which also only serves the `@delon/*` class library), **you don't use in your project**. If your project need to support i18n, then using [$localize](https://angular.io/api/localize/init/$localize).

Or you can use NG-ALAIN built-in simple i18n solution, pls refer to [ALAIN_I18N_TOKEN](https://ng-alain.com/docs/i18n/en?#ALAIN_I18N_TOKEN).

See also:
- https://angular.io/guide/i18n-overview
